### DISCUSSION TITLE
Could you please tell me about the context of the extension?

### DISCUSSION THREAD

**Discussion by myyrakle** [tokio-rs/axum#1571] [Q&A] [answered]
For example I
1. Database connection object
2. Authentication-related values ​​of the current request

I want to manage the above two as state values.

Number 1 is a DB setting for all handlers or middleware to share and use.
Number 2 is to read the authorization in the middleware and pass the authorization result to the handler.

In my opinion, number 1 should be server-wide state sharing, and number 2 should be state sharing specific to a specific request.

Can axum Extension manage these two separately? What are the default settings?

**Comment by davidpdrsn** [comment, marked as answer]
> Can axum Extension manage these two separately?

Yes. Do `router.layer(Extension(db))` for the global and from your auth middleware do `request.extensions_mut().insert(...)`.

**Reply by myyrakle** [reply]
Thank you for quick response!
